File: LeetCode/Top Interview 150/[240725] 121. Best Time to Buy and Sell Stock/정훈.go
func maxProfit(prices []int) int {
    minPrice := prices[0]
    maxProfit := 0
    for _, p := range prices {
        minPrice = min(minPrice, p)
        maxProfit = max(maxProfit, p - minPrice)
    }
    return maxProfit
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

func min(a, b int) int {
    if a > b {
        return b
    }
    return a
}

File: LeetCode/Top Interview 150/[240718] 122. Best Time to Buy and Sell Stock II/정훈.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # 팔거나 사거나, 안사거나
        bought = False
        profit = 0
        for i in range(len(prices) - 1):
            if not bought:
                if prices[i+1] > prices[i]:
                    bought = True
                    bPrice = prices[i]
            else:
                if prices[i+1] <= prices[i]:
                    bought = False
                    profit += prices[i]-bPrice
        if bought:
            profit += prices[-1]-bPrice
        return profit
